### ISSUE TITLE
about touchmove, touchend listeners

### ISSUE THREAD

**Issue by Shoyi** [chakming/HTML5-Chinese-Calligraphy#3]
there is no removelistener for "touchmove" and "touchend" in this program.
this issue caused that the ink might not smoothly flow in mobile .

**Comment by pedoc** [comment]
yes,i also too.
i fixed,work fine
```
init: function (canvasElement, updateCallback, offset) {
        //alert(JSON.stringify(offset));
        this._offset=offset;
        this._canvasElement = canvasElement,
            this._width = canvasElement.width,
            this._height = canvasElement.height,
            this._updateCallback = updateCallback,
            this._initEventHandlers(),
            this._prevMousePoint = [0, 0],
            this._prevSmoothMousePoint = [0, 0],
            this._prevMouseChangeVector = [0, 0],
            this._mouseSmoothingFactor = 0.3,
            this._lastRotation = Math.PI / 2,
            this._ctx = this._canvasElement.getContext('2d'),
            this._config.oldFillStyle = this._ctx.fillStyle,
            this._config.oldLineWidth = this._ctx.lineWidth,
            this.isDrawing = false,
            this.penStyle = 'art',
            this.easerPenWidth=40,
            this.eventBinded = false; //there
    }
```
and
```
 _onDrawStart: function (a) { // == variable 'a': event == //
        //console.log('_onDrawStart',(new Date).getTime());
        //console.log('_onDrawStart', 'penStyle=', this.penStyle,'a=',a);
        this.isDrawing=true;
        a.preventDefault(), a.stopPropagation(), a.touches && (a = a.touches[0]);
        if (this._activeTool instanceof Tool) {
            
            var c = a.pageX - this._canvasElement.offsetLeft,
                d = a.pageY - this._canvasElement.offsetTop;
      
            this._resetMouseSmoothing(c , d),
                this._penStyleChanged(c - this._offset.x, d - this._offset.y, a.pageX, a.pageY),
                this._activeTool.mousedown(c, d, a.pageX, a.pageY);
            if (!this.eventBinded) {//if eventhandler is bind,nothing to do

                // todo: jquery!
                $(this._canvasElement).bind('mousemove', delegate(this, this._onDrawMove), false).bind('mouseup', delegate(this, this._onDrawEnd), false);

                //$(this._canvasElement).bind('touchmove.Draw',delegate(this, this._onDrawMove), false),
                //$(this._canvasElement).bind('touchend.Draw',delegate(this, this._onDrawEnd), false);
                this._canvasElement.addEventListener('touchmove', delegate(this, this._onDrawMove), false);

                this._canvasElement.addEventListener('touchend', delegate(this, this._onDrawEnd), false);

                //var e = document.getElementsByTagName('body')[0];
                //$('body').bind('mouseup.Draw',delegate(this, this._onDrawEnd), false);
                //$('body').addEventListener('touchend',delegate(this, this._onDrawEnd), false);

                this.eventBinded = true;
            }
        }
    }

```

**Comment by chakming** [comment]
welcome PR :)

**Comment by chakming** [comment]
merged #4
